src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -103,9 +103,9 @@ public void IncomingComplete(int count, Exception error)
                 {
                     FinReceived();
                 }
-
-                Complete();
             }
+
+            Complete();
```

Comment:
Since you are in a lock could just change them to an if statement rather than `Interlocked.CompareExchange`?